sonic-thermalctld/scripts/thermalctld

```diff
@@ -718,97 +731,118 @@ class ThermalControlDaemon(daemon_base.DaemonBase):
+            sys.exit(ERR_INIT_FAILED)
+
+    def deinit(self):
+        """
+        Deinitializer of ThermalControlDaemon
+        """
+        try:
+            if self.thermal_manager:
+                self.thermal_manager.deinitialize()
+        except Exception as e:
+            self.log_error('Caught exception while destroying thermal manager - {}'.format(e))
+
+        self.thermal_monitor.task_stop()
+
+    # Override signal handler from DaemonBase
     def signal_handler(self, sig, frame):
         """
         Signal handler
         :param sig: Signal number
         :param frame: not used
         :return:
         """
-        if sig == signal.SIGHUP:
-            self.log_info("Caught SIGHUP - ignoring...")
-        elif sig == signal.SIGINT or sig == signal.SIGTERM:
-            self.log_info("Caught signal {} - exiting...".format(sig))
-            self.exit_code = sig + 128
+        FATAL_SIGNALS = [signal.SIGINT, signal.SIGTERM]
+        NONFATAL_SIGNALS = [signal.SIGHUP]
+
+        global exit_code
+
+        if sig in FATAL_SIGNALS:
+            self.log_info("Caught signal '{}' - exiting...".format(SIGNALS_TO_NAMES_DICT[sig]))
+            exit_code = 128 + sig  # Make sure we exit with a non-zero code so that supervisor will try to restart us
             self.stop_event.set()
+        elif sig in NONFATAL_SIGNALS:
+            self.log_info("Caught signal '{}' - ignoring...".format(SIGNALS_TO_NAMES_DICT[sig]))
         else:
-            self.log_warning("Caught unhandled signal '" + sig + "'")
+            self.log_warning("Caught unhandled signal '{}' - ignoring...".format(SIGNALS_TO_NAMES_DICT[sig]))
 
+    # Main daemon logic
     def run(self):
         """
         Run main logical of this daemon
         :return:
         """
-        self.log_info("Starting up...")
-
-        import sonic_platform.platform
-        chassis = sonic_platform.platform.Platform().get_chassis()
-
-        thermal_monitor = ThermalMonitor(chassis)
-        thermal_monitor.task_run()
+        if self.stop_event.wait(self.wait_time):
+            # We received a fatal signal
+            return False
 
-        thermal_manager = None
+        begin = time.time()
         try:
-            thermal_manager = chassis.get_thermal_manager()
-            if thermal_manager:
-                thermal_manager.initialize()
-                thermal_manager.load(ThermalControlDaemon.POLICY_FILE)
-                thermal_manager.init_thermal_algorithm(chassis)
-        except NotImplementedError:
-            self.log_warning('Thermal manager is not supported on this platform.')
+            if self.thermal_manager:
+                self.thermal_manager.run_policy(chassis)
         except Exception as e:
-            self.log_error('Caught exception while initializing thermal manager - {}'.format(repr(e)))
+            self.log_error('Caught exception while running thermal policy - {}'.format(e))
```

Comment:
The problem is `repr(e)` produces slightly different output between Python 2 and Python 3 (one ends with a comma, the other doesn't), whereas the output of `e` is identical. If we didn't have to continue supporting both versions, it wouldn't be a problem. Without this change, I will need to add ugly version checks in the unit tests. I can do it if you insist, and we can simply clean it all up once we no longer support Python 2.

Reply:
Understood. Then repr(e) is not necessary. As long as the exception is reproduceable, we can manually add it for debug.

Reply:
@Junchao-Mellanox: I went back and changed all log statements to print `repr(e)`, seeing as I'd rather make the unit tests temporarily messy than to potentially make debugging more difficult. Since we'll soon remove Python 2 support, this will be a short-lived inconvenience, and only if modifying the unit tests. Please review again.